### ISSUE TITLE
Get the value of loss function during training.

### ISSUE THREAD

**Issue by DaniiltheGreat** [tensorforce/tensorforce#820]
Good morning, everybody.

I have a bit silly question. I am using the Act-observe-Environment interface as proposed in examples to train different agents.     My question is: how can I retreive the values of the loss function? Is it possible to check loss function values inside the training loop?

When I try to read summaries, all values for all saved parameters (nor only loss) are equal to zero. Meanwhile the agent learns to make optimal actions, so is trained.
Example of agent:

> agent_rat = Agent.create(
    agent='dqn',
    # Envorinment
    actions = {'type': 'int', 'shape': (), 'num_values': 2},#spaces.MultiDiscrete([8, 2]),
    states = {'type': 'float', 'shape': (2,), 'min_value': None, 'max_value': None},
    #{'type': 'int', 'shape': (2,), 'num_values': 15}, #, 'min_value': , 'max_value':
    #spaces.Box(low=-np.inf, high=np.inf, shape=(2,), dtype=np.float32), #0r shape = (2,),
    # Automatically configured network
    network=
    [
        dict(type='dense', size=64, activation='leaky-relu', bias=True),
        dict(type='dense', size=64, activation='leaky-relu', bias=True),
        dict(type='dense', size=64, activation='leaky-relu', bias=True),
        dict(type='dense', size=64, activation='leaky-relu', bias=True)
    ],
    # optimization parameters
    batch_size=50,
    update_frequency=1.0,
    learning_rate=0.1,
    # optimizer='adam',
    # subsampling_fraction=0.33,
    # Reward estimation
    discount=1.0,
    predict_terminal_values=False,
    horizon=1,
    # Baseline network and optimizer
    memory=53,
    # baseline=dict(type='auto', size=8, depth=1),
    # baseline_optimizer=dict(optimizer='adam', learning_rate=0.1),
    # Regularization
    l2_regularization=0.0, entropy_regularization=0.0,
    # Preprocessing
    state_preprocessing=None,
    reward_preprocessing=None,
    # Exploration
    exploration=0.05,
    variable_noise=0.0,
    # Default additional config values
    config=None,
    # Save agent every 10 updates and keep the 5 most recent checkpoints
    saver=dict(directory='./model_rat', frequency=5, max_checkpoints=5),
    # Log all available Tensorboard summaries
    summarizer=dict(directory='./summaries_rat',
                    summaries=['all', 'entropy', 'kl-divergence', 'loss', 'reward', 'update-norm']),
    # Do not record agent-environment interaction trace
    tracking="all",
    recorder=None
)

How I read summaries:

> for e in tf.compat.v1.train.summary_iterator(summary_dir):
        for v in e.summary.value:
                if v.tag == 'agent/cond/losses/policy-objective-loss':
                        print(v.simple_value)

My tensorforce  version is 0.6.4.

Thank you in advance.

**Comment by AlexKuhnle** [comment]
Hi @DaniiltheGreat,

the `summarizer` argument is more for Tensorboard visualisation, so what you want is provided by the `tracking` argument. You've already set that to `all`, so you can get the latest tensor values by calling `agent.tracked_tensors()`  after each agent call. Note that not all values are used/updated in every agent call, in fact most values (in particular the loss) is only updated as part of an observe call which involved an update (i.e. if `agent.observe(...)` returns `True`). 

Hope that helps!

**Comment by DaniiltheGreat** [comment]
Yes, thank you. I misunderstood description.
